src/storage/table/column_segment.cpp

```diff
@@ -483,6 +484,19 @@ idx_t ColumnSegment::FilterSelection(SelectionVector &sel, Vector &result, const
 		return TemplatedNullSelection<true>(sel, approved_tuple_count, mask);
 	case TableFilterType::IS_NOT_NULL:
 		return TemplatedNullSelection<false>(sel, approved_tuple_count, mask);
+	case TableFilterType::STRUCT_EXTRACT: {
+		auto &struct_filter = filter.Cast<StructFilter>();
+		auto &child_vec = StructVector::GetEntries(result)[struct_filter.child_idx];
+
+		// Also do a null-scan here
+		// This is safe (and neccessary), because we ONLY produce StructFilters when theres a constant comparison on the
+		// struct field which must always have a IS NOT NULL clause applied anyway.
+		TemplatedNullSelection<false>(sel, approved_tuple_count, FlatVector::Validity(*child_vec));
```

Comment:
Can we remove this and instead use a conjunction with a `NullFilter` similar to how the top-level filter pushdown works?